src/cargo/sources/path.rs

```diff
@@ -226,7 +226,8 @@ pub struct RecursivePathSource<'gctx> {
     /// Whether this source has loaded all package information it may contain.
     loaded: bool,
     /// Packages that this sources has discovered.
-    packages: HashMap<PackageId, Package>,
+    packages: HashMap<PackageId, Vec<Package>>,
```

Comment:
Could we document what the key-value pair for? I can see it a bit unintuitive because `RecurisivePathSource` ignores all but the first.